devtools/server/actors/replay/connection.js

```diff
@@ -533,6 +527,114 @@ class Recording extends EventEmitter {
   }
 }
 
+function checkShouldValidateUrl() {
+  if (gShouldValidateUrl !== null) {
+    return Promise.resolve(gShouldValidateUrl)
+  } else {
+    return queryAPIServer(`
+      query GetOrgs {
+        viewer {
+          workspaces {
+            edges {
+              node {
+                isOrganization
+                settings {
+                  features
+                }
+              }
+            }
+          }
+        }
+      }
+    `).then(resp => {
+      if (resp.errors) {
+        throw new Error("Unexpected error checking Replay user permissions");
+      }
+
+      const workspaces = resp.data.viewer.workspaces.edges;
+      gShouldValidateUrl = !workspaces ? false : workspaces.some(w => {
+        if (w.node.isOrganization) {
+          const {allowList, blockList} = w.node.settings?.features?.recording || {};
+
+          return (Array.isArray(allowList) && allowList.length > 0) || (Array.isArray(blockList) && blockList.length > 0);
```

Comment:
I agree. I added this check to avoid calling out to check for users that aren't in an org (the majority) since it feels a bit too "big brother" to ping on every URL (even though we're only doing it while recording so not really any difference).

To remedy this, we can either:
* Accept that every URL will be routed through the backend to check it, or
* Add another graphql element that reports if allow/block listing is enabled

The latter is easy enough to do though if we switch to this being handled by the protocol, it'll likely become mute since we'll probably need to the former then.

Reply:
Yeah definitely a conversation to have. We could also just add an endpoint to fetch the URL patterns, and then implement all the allowlist/blocklist validation in the browser itself, which could avoid a lot of that roundtripping.

Reply:
Sure but then we wouldn't need the protocol change but we'd still need the graphql call :) Regardless, we'll address it in follow up.

Reply:
Oh I was figuring you'd still use the protocol to ask for the set of allow/block patterns.